core/server/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1912,14 +1912,14 @@ public synchronized FileSystemCommand workerHeartbeat(long workerId, List<Long>
    * @param options the method options
    * @throws FileDoesNotExistException
    */
-  private void setStateInternal(long fileId, long opTimeMs, SetAttributeOptions options)
+  private void setAttributeInternal(long fileId, long opTimeMs, SetAttributeOptions options)
       throws FileDoesNotExistException {
     Inode inode = mInodeTree.getInodeById(fileId);
-    if (options.hasPinned()) {
+    if (options.getPinned() != null) {
```

Comment:
because the idiom we are currently using for the client-side and server-side options is different ... it sucks and I would like to fix that, but I would prefer to do it as a follow up to this PR, because it is going to be a lot of churn